operator/v1/types_machineconfiguration.go

```diff
@@ -57,8 +57,39 @@ type MachineConfigurationSpec struct {
 }
 
 type MachineConfigurationStatus struct {
-	// TODO tombstone this field
-	StaticPodOperatorStatus `json:",inline"`
+	// observedGeneration is the last generation change you've dealt with
```

Comment:
What does this field actually mean to the MachineConfiguration?

Reply:
I think our thought was that this will be used by consumers of Status, to ensure that it's working on the latest user configuration. 
For example, our current workflow for NodeDisruptionPolicy works like this 
- user applies a MachineConfiguration object with some node disruption fields set
- the operator see this, merges the NodeDisruptionPolicy spec with the MCO defaults and updates the NodeDisruptionStatus
- during an update, the MCO's daemonset will look at the Status and apply the policies on the update diff.
For the final step, there is no version/generation check by the daemonset currently; as long as a Status exists, the daemon will attempt to use it. With an ObservedGeneration, we can possibly gate on it being up to date before using the policy.